zipline/finance/risk/cumulative.py

```diff
@@ -312,10 +312,18 @@ def update(self, dt, algorithm_returns, benchmark_returns, account):
         self.excess_returns[self.latest_dt] = (
             self.algorithm_cumulative_returns[self.latest_dt] -
             self.treasury_period_return)
-        self.metrics.beta[dt] = self.calculate_beta()
+        dt_index = self.metrics.index.searchsorted(dt)
+        algorithm_returns = self.algorithm_returns.values[:dt_index + 1]
```

Comment:
Does this slice into the array produce the same effect as the dropna call in calculate_beta (but without the memory copy)?

If so, is it ok that we're calling calculate_downside_risk with that memview (and similarly with the mean_returns) now, whereas we were not dropping nans in calculate_downside_risk before?